packages/mui-material-next/src/InputBase/InputBase.tsx

```diff
@@ -80,25 +81,30 @@ const useUtilityClasses = (ownerState: InputBaseOwnerState) => {
   return composeClasses(slots, getInputBaseUtilityClass, classes);
 };
 
+export function rootOverridesResolver(
+  props: InputBaseRootSlotProps,
+  styles: Record<string, CSSInterpolation>,
+) {
+  const { ownerState } = props;
+
+  return [
+    styles.root,
+    ownerState.formControl && styles.formControl,
+    ownerState.startAdornment && styles.adornedStart,
+    ownerState.endAdornment && styles.adornedEnd,
+    ownerState.error && styles.error,
+    ownerState.size === 'small' && styles.sizeSmall,
+    ownerState.multiline && styles.multiline,
+    ownerState.color && styles[`color${capitalize(ownerState.color)}`],
+    ownerState.fullWidth && styles.fullWidth,
+    ownerState.hiddenLabel && styles.hiddenLabel,
+  ];
+}
+
 export const InputBaseRoot = styled('div', {
   name: 'MuiInputBase',
   slot: 'Root',
-  overridesResolver: (props, styles) => {
-    const { ownerState } = props;
-
-    return [
-      styles.root,
-      ownerState.formControl && styles.formControl,
-      ownerState.startAdornment && styles.adornedStart,
-      ownerState.endAdornment && styles.adornedEnd,
-      ownerState.error && styles.error,
-      ownerState.size === 'small' && styles.sizeSmall,
-      ownerState.multiline && styles.multiline,
-      ownerState.color && styles[`color${capitalize(ownerState.color)}`],
-      ownerState.fullWidth && styles.fullWidth,
-      ownerState.hiddenLabel && styles.hiddenLabel,
-    ];
-  },
+  overridesResolver: rootOverridesResolver,
```

Comment:
I like this refactor of moving the overridesResolver out. Maybe something we could apply everywhere.

Reply:
It was actually like this to begin with, I accidentally un-exported it when I was working on `InputBase` 🙈